src/framework/preferences/js/SeparatedPanelPrefsEditor.js

```diff
@@ -330,7 +339,84 @@ var fluid_2_0_0 = fluid_2_0_0 || {};
+        fluid.invokeLater(callback);
+    };
+
+    fluid.defaults("fluid.prefs.separatedPanel.lazyLoad", {
+        events: {
+            onLazyLoad: null,
+            onPrefsEditorMessagesPreloaded: null,
+            onCreateSlidingPanelReady: {
+                events: {
+                    onPrefsEditorMessagesLoaded: "onPrefsEditorMessagesPreloaded"
+                }
+            },
+            templatesAndIframeReady: {
+                events: {
+                    onLazyLoad: "onLazyLoad"
+                }
+            }
+        },
+        components: {
+            templateLoader: {
+                createOnEvent: "onLazyLoad"
+            },
+            messageLoader: {
+                options: {
+                    events: {
+                        onResourcesPreloaded: "{separatedPanel}.events.onPrefsEditorMessagesPreloaded"
+                    },
+                    preloadResources: "prefsEditor",
+                    listeners: {
+                        "onCreate.loadResources": {
+                            listener: "fluid.prefs.separatedPanel.lazyLoad.preloadResources",
+                            args: ["{that}", {expander: {func: "{that}.resolveResources"}}, "{that}.options.preloadResources"]
+                        },
+                        "{separatedPanel}.events.onLazyLoad": {
+                            listener: "fluid.resourceLoader.loadResources",
+                            args: ["{messageLoader}", {expander: {func: "{messageLoader}.resolveResources"}}]
+                        }
+                    }
+                }
+            },
+            slidingPanel: {
+                options: {
+                    invokers: {
+                        operateShow: {
+                            funcName: "fluid.prefs.separatedPanel.lazyLoad.showPanel",
+                            args: ["{separatedPanel}", "{that}.events.afterPanelShow.fire"]
+                        }
+                    }
+                }
+            }
+        }
+    });
+
+    fluid.prefs.separatedPanel.lazyLoad.showPanel = function (separatedPanel, callback) {
+        if (separatedPanel.prefsEditor) {
+            fluid.invokeLater(callback);
+        } else {
+            separatedPanel.events.onReady.addListener(function (that) {
+                that.events.onReady.removeListener("showPanelCallBack");
+                fluid.invokeLater(callback);
+            }, "showPanelCallBack");
+            separatedPanel.events.onLazyLoad.fire();
+        }
+
+    };
+
+    fluid.prefs.separatedPanel.lazyLoad.preloadResources = function (that, resources, toPreload) {
+        toPreload = fluid.makeArray(toPreload);
+        var preloadResources = {};
+
+        fluid.each(toPreload, function (resourceName) {
+            preloadResources[resourceName] = resources[resourceName];
+        });
+
+        fluid.fetchResources(preloadResources, function () {
+            that.resources = preloadResources;
```

Comment:
This is part of a function that is used to reconfigure how the messageLoader fetches resources. This is because we need to load in some of the messages, i.e the text on the tab of the sliding panel, but don't want to load in all of the strings for the panels until the prefsEditor is to be instantiated. That is why the tests verifies `prefsEditorLoader.messageLoader.resources`.